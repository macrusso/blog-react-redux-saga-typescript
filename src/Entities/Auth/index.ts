import * as authConstants from './authConstants';
import * as authActions from './authActions';
import * as authSagas from './authSagas';
import authReducer, { ICommentsState } from './authReducer';

export { authSagas, ICommentsState, authActions, authReducer, authConstants };
