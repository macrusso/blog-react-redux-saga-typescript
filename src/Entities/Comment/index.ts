import * as commentConstants from './commentConstants';
import * as commentSelectors from './commentSelectors';
import * as commentActions from './commentActions';
import * as commentSagas from './commentSagas';
import commentReducer, { ICommentsState } from './commentReducer';
import ICommentPartial from './ICommentPartial';
import IComment from './IComment';

export {
  IComment,
  commentSagas,
  ICommentsState,
  commentActions,
  commentReducer,
  ICommentPartial,
  commentSelectors,
  commentConstants,
};
