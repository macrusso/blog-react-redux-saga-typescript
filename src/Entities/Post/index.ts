import * as postConstants from "./postConstants";
import * as postSelectors from "./postSelectors";
import * as postActions from "./postActions";
import * as postSagas from "./postSagas";
import postReducer from "./postReducer";
import Post from "../../Frontend/Post/PostListItem";
import IPostPartial from "./IPostPartial";
import IPost from "./IPost";
import IPostsState from "./IPostsState";

export { Post, IPost, postSagas, postActions, IPostsState, postReducer, IPostPartial, postSelectors, postConstants };
