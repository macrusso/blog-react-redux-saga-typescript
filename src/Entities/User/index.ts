import * as userConstants from './userConstants';
import * as userSelectors from './userSelectors';
import * as userActions from './userActions';
import * as userSagas from './userSagas';
import userReducer, { IUsersState } from './userReducer';
import IUserAuth from './IUserAuth';
import IUserPartial from './IUserPartial';
import IUser from './IUser';

export {
  IUser,
  IUserAuth,
  userSagas,
  userActions,
  IUsersState,
  userReducer,
  IUserPartial,
  userSelectors,
  userConstants,
};
