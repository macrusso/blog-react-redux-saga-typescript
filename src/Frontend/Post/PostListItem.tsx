import React from 'react';
import { Item, Container } from 'semantic-ui-react';
import { IPost } from '../../Entities/Post';
import { IUser } from '../../Entities/User';
import { Link } from 'react-router-dom';
import { selectedPost } from '../../routes';

export interface IPostListItemProps {
  posts: IPost[];
  users: IUser[];
  loading: boolean;
  usersLoading: boolean;
}

const PostListItem = ({
  users,
  posts,
  loading,
  usersLoading,
}: IPostListItemProps) => (
  <Container>
    <Item.Group>
      {!loading &&
        !usersLoading &&
        posts.length > 0 &&
        posts.map(post => (
          <Item key={post.id}>
            <Item.Image
              size="tiny"
              src="https://react.semantic-ui.com/images/wireframe/image.png"
            />
            <Item.Content>
              <Item.Header>
                <Link to={selectedPost.replace(':postId', post.id.toString())}>
                  {post.title}
                </Link>
              </Item.Header>
              <Item.Meta>by {users[post.userId].name}</Item.Meta>
              <Item.Description>{post.body}</Item.Description>
            </Item.Content>
          </Item>
        ))}
    </Item.Group>
  </Container>
);

export default PostListItem;
