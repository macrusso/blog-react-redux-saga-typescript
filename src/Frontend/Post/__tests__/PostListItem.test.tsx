import React from "react";
import { BrowserRouter } from "react-router-dom";
import { shallow } from "enzyme";
import PostListItem from "../PostListItem";
import { IPost, IUser, IComment } from "../../../Entities";

describe("PostListItem", () => {
  test("Snapshot matches", () => {
    const posts: IPost[] = [
      {
        _id: "5bcc554259e0dbfda6ed1a55",
        userId: "5bcc554259e0dbfda6ed1a55",
        title: "string",
        body: "string",
      },
    ];

    const users: { [key: string]: IUser } = {
      "5bcc554259e0dbfda6ed1a55": {
        name: "string",
        email: "string",
        _id: "5bcc554259e0dbfda6ed1a55",
      },
    };

    const comments: IComment[] = [
      {
        body: "string",
        _id: "5bcc554259e0dbfda6ed1a55",
        postId: "5bcc554259e0dbfda6ed1a55",
        userId: "5bcc554259e0dbfda6ed1a55",
      },
    ];
    const wrapper = shallow(
      <BrowserRouter>
        <PostListItem
          posts={posts}
          users={users}
          comments={comments}
          loading={false}
          usersLoading={false}
          onAddCommentClick={jest.fn()}
        />
      </BrowserRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
