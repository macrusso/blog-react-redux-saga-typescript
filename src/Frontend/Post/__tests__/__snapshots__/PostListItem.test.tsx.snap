// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListItem Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserRouter>
    <WithStyles(PostListItem)
      loading={false}
      posts={Array []}
      users={Array []}
      usersLoading={false}
    />
  </BrowserRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(PostListItem)
        loading={false}
        posts={Array []}
        users={Array []}
        usersLoading={false}
      />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "loading": false,
        "posts": Array [],
        "users": Array [],
        "usersLoading": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(PostListItem)
          loading={false}
          posts={Array []}
          users={Array []}
          usersLoading={false}
        />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "loading": false,
          "posts": Array [],
          "users": Array [],
          "usersLoading": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
